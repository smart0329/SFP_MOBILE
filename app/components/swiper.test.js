// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import Swiper from './swiper.js';

describe('Swiper', () => {
    const baseProps = {
        children: [],
    };

    test('should match snapshot', async () => {
        const wrapper = shallow(
            <Swiper {...baseProps}/>,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should not call scrollView.scrollTo and updateIndex when children < 2', () => {
        const children = [];
        while (children.length < 2) {
            const props = {...baseProps, children};
            const wrapper = shallow(
                <Swiper {...props}/>,
            );

            const instance = wrapper.instance();
            instance.updateIndex = jest.fn();
            instance.refScrollView({
                scrollTo: jest.fn(),
            });

            instance.scrollToIndex(1, false);
            expect(instance.scrollView.scrollTo).not.toHaveBeenCalled();
            expect(instance.updateIndex).not.toHaveBeenCalled();

            children.push('test');
        }

        expect(children.length).toBe(2);
    });

    test('should call scrollView.scrollTo and updateIndex when children >= 2', () => {
        const children = [1, 2, 3];
        while (children.length >= 2) {
            const props = {...baseProps, children};
            const wrapper = shallow(
                <Swiper {...props}/>,
            );

            const instance = wrapper.instance();
            instance.updateIndex = jest.fn();
            instance.refScrollView({
                scrollTo: jest.fn(),
            });

            instance.scrollToIndex(1, false);
            expect(instance.scrollView.scrollTo).toHaveBeenCalled();
            expect(instance.updateIndex).toHaveBeenCalled();

            children.pop();
        }

        expect(children.length).toBe(1);
    });
});
