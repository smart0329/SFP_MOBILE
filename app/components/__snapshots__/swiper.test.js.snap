// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swiper should match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "position": "relative",
      },
    ]
  }
>
  <ScrollView
    automaticallyAdjustContentInsets={true}
    bounces={false}
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    horizontal={true}
    keyboardShouldPersistTaps="handled"
    onMomentumScrollEnd={[Function]}
    onScrollBeginDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={true}
    scrollEnabled={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
  />
</View>
`;
